opennms-dao/src/main/java/org/opennms/netmgt/dao/hibernate/OutageDaoHibernate.java

```diff
@@ -103,6 +111,76 @@ public OnmsOutage currentOutageForService(OnmsMonitoredService service) {
         });
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public Collection<OutageDetails> newestOutages(final List<String> serviceNames) {
+        return getHibernateTemplate().execute(new HibernateCallback<List<OutageDetails>>() {
+            @Override
+            @SuppressWarnings("unchecked")
+            public List<OutageDetails> doInHibernate(Session session) throws HibernateException, SQLException {
+                final StringBuilder query = new StringBuilder()
+                        .append("SELECT DISTINCT\n")
+                        .append("        outages.outageId,\n")
+                        .append("        outages.ifServiceId AS monitoredServiceId,\n")
+                        .append("        service.serviceName AS serviceName,\n")
+                        .append("        outages.ifLostService,\n")
+                        .append("        outages.ifRegainedService,\n")
+                        .append("        node.nodeId,\n")
+                        .append("        node.foreignSource,\n")
+                        .append("        node.foreignId,\n")
+                        .append("        node.location\n")
+                        .append("FROM outages\n")
+                        .append("        LEFT JOIN ifServices ON outages.ifServiceId = ifServices.id\n")
+                        .append("        LEFT JOIN service ON ifServices.serviceId = service.serviceId\n")
+                        .append("        LEFT JOIN ipInterface ON ifServices.ipInterfaceId = ipInterface.id\n")
+                        .append("        LEFT JOIN node ON ipInterface.nodeId = node.nodeId\n")
+                        .append("        LEFT JOIN (\n")
+                        .append("                SELECT DISTINCT\n")
```

Comment:
This statement still ends up doing a sequential scan of the whole outages table.

I'm wondering if we really need to obtain the last outage record for resolved outages? If we only select those that are still open this query becomes much simpler and possible to do with HQL / criteria. The lack of the record implies that the outage was resolved.